src/vampir/vampir.lisp

```diff
@@ -68,3 +68,256 @@
+                             :rhs (make-infix :op :*
+                                              :lhs (make-constant :const (expt 2 n))
+                                              :rhs (make-wire :var (intern
+                                                                    (format nil "A~a" n) :keyword)))))))
+    (labels ((in-tuple (n)
+               (if (= n 0)
+                   (make-wire :var :a0)
+                   (make-tuples :wires (list
+                                        (in-tuple (- n 1))
+                                        (make-wire :var (intern
+                                                         (format nil "A~a" n) :keyword)))))))
+      (make-alias
+       :name :range32
+       :inputs (list :a)
+       :body
+       (append (mapcar
+                (lambda (x y)
+                  (spc:make-bind :names (list (make-wire :var y)) :value
+                                 (make-application :func :fresh
+                                                   :arguments (list (make-constant :const x)))))
+                (mapcar (lambda (x) (expt 2 x))
+                        (alexandria:iota 32 :start 1))
+                (mapcar (lambda (x) (intern
+                                     (format nil "A~a" x) :keyword))
+                        (alexandria:iota 32)))
+               (list (make-equality :lhs (make-wire :var :a)
+                                    :rhs (in-plus 31)))
+               (list (make-tuples :wires (list
+                                          (in-tuple 31)
+                                          (make-wire :var :|()|)))))))))
+
+(defun range32 (a)
+  (make-application :func :range32
+                    :arguments (list a)))
+
+(defparameter *intrange32*
+  (make-alias :name :intRange32
+              :inputs (list :a)
+              :body (list (make-application :func :range32
+                                            :arguments (list (make-infix :op :+
+                                                                         :lhs (make-wire :var :a)
+                                                                         :rhs (make-constant :const 2147483648)))))))
+
+(defun intrange32 (a)
+  (make-application :func :intrange32
+                    :arguments (list a)))
+
+
+(defparameter *negative32*
+  (flet ((wires (num)
+           (make-wire :var (intern (format nil "A~a" num) :keyword))))
+    (make-alias
+     :name   :negative32
+     :inputs (list :a)
+     :body
+     (list (make-bind
+            :names (append (mapcar #'wires (alexandria:iota 32))
+                           (list (make-wire :var :|()|)))
+            :value (make-application
+                    :func      :intrange32
+                    :arguments (list (make-wire :var :a))))
+           (make-wire :var :a31)))))
+
+(defun negative32 (a)
+  (make-application :func :negative32
+                    :arguments (list a)))
+
+(defparameter *nonnegative32*
```

Comment:
for reading usefulness, put dasshes

```lisp
non-negative32
```

reads much nicer than

```lisp
nonnegative32
```
I almost typed that skipping a `n`